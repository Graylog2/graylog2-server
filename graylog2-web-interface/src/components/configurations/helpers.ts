const getConfig = (configType, configuration) => configuration?.[configType] ?? null;

export { getConfig };
export default getConfig;
