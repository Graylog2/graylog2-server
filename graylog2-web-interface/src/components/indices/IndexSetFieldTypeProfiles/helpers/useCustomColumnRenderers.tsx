/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React, { useMemo } from 'react';

import type { CustomFieldMapping, IndexSetFieldTypeProfile } from 'components/indices/IndexSetFieldTypeProfiles/types';
import CustomFieldMappingsCell from 'components/indices/IndexSetFieldTypeProfiles/cells/CustomFieldMappingsCell';
import IndexSetsCell from 'components/indices/IndexSetFieldTypeProfiles/cells/IndexSetsCell';

const useCustomColumnRenderers = (normalizedIndexSetsTitles: Record<string, string>) => useMemo(() => ({
  attributes: {
    custom_field_mappings: {
      renderCell: (customFieldTypes: Array<CustomFieldMapping>, profile: IndexSetFieldTypeProfile) => (
        <CustomFieldMappingsCell profile={profile}
                                 customFieldTypes={customFieldTypes} />
      ),
      staticWidth: 200,
    },
    index_set_ids: {
      renderCell: (indexSetIds: Array<string>) => <IndexSetsCell indexSetIds={indexSetIds} normalizedIndexSetsTitles={normalizedIndexSetsTitles} />,
    },
  },
}), [normalizedIndexSetsTitles]);

export default useCustomColumnRenderers;
