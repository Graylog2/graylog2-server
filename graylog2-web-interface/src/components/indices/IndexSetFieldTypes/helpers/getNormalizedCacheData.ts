/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */

import type { Query } from '@tanstack/react-query';
import keyBy from 'lodash/keyBy';
import reduce from 'lodash/reduce';
import extend from 'lodash/extend';

import type { IndexSetFieldTypesQueryData, IndexSetFieldType } from 'components/indices/IndexSetFieldTypes/types';

const getNormalizedCacheData = (queryCache: Array<Query<IndexSetFieldTypesQueryData>>): Record<string, IndexSetFieldType> => {
  const cacheData: Array<Record<string, IndexSetFieldType>> = queryCache.map(({ state }) => keyBy(state?.data?.list, 'id'));
  const mergedCache: Record<string, IndexSetFieldType> = reduce<Record<string, IndexSetFieldType>>(cacheData, extend);

  return mergedCache;
};

export default getNormalizedCacheData;
