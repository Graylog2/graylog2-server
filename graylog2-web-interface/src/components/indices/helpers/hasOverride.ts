import type { IndexSetFieldType } from 'components/indices/IndexSetFieldTypes/hooks/useIndexSetFieldType';

const hasOverride = (fieldType: IndexSetFieldType) => fieldType.origin === 'OVERRIDDEN_PROFILE' || fieldType.origin === 'OVERRIDDEN_INDEX';

export default hasOverride;
