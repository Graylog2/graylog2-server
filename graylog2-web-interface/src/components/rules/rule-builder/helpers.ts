/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import type { BlockDict, RuleBuilderRule } from './types';

const getDictForFunction = (dict: BlockDict[], functionName: string) : BlockDict | undefined => (
  dict?.find((entry) => entry.name === functionName)
);

const jsonifyText = (text: string): string => {
  try {
    JSON.parse(text);

    return text;
  } catch {
    try {
      const rawMessageToJson = `{"${
        text
          .trim()
          .replace(/^\s*\n/gm, '')
          .replace(/"/g, '\\"')
          .split('\n')
          .map((line) => line.trim().split(':').map((keyValue) => keyValue.trim()))
          .filter((keyValue) => keyValue[0] && keyValue[1])
          .map((keyValue) => `${keyValue[0].replace(/\\"/g, '').trim()}":"${keyValue.slice(1).join(':').trim()}`)
          .join('","')
      }"}`;

      JSON.parse(rawMessageToJson);

      return rawMessageToJson;
    } catch {
      return JSON.stringify({ message: text });
    }
  }
};

const hasRuleBuilderErrors = (rule: RuleBuilderRule): boolean => {
  if (rule?.rule_builder?.errors?.length > 0) return true;

  if (rule?.rule_builder?.actions?.some(((action) => action.errors?.length > 0))) {
    return true;
  }

  if (rule?.rule_builder?.conditions?.some(((condition) => condition.errors?.length > 0))) {
    return true;
  }

  return false;
};

export { getDictForFunction, jsonifyText, hasRuleBuilderErrors };
