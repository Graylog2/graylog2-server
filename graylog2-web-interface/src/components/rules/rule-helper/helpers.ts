/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
const niceType = (typeName) => typeName.replace(/^.*\.(.*?)$/, '$1');

const functionSignature = (descriptor) => {
  const args = descriptor.params.map((p) => (p.optional ? `[${p.name}]` : p.name));

  return `${descriptor.name}(${args.join(', ')}) : ${niceType(descriptor.return_type)}`;
};

export { functionSignature, niceType };
