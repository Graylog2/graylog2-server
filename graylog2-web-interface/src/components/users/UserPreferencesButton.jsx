import PropTypes from 'prop-types';
import React from 'react';
import { Button } from 'react-bootstrap';

import UserPreferencesModal from 'components/users/UserPreferencesModal';

class UserPreferencesButton extends React.Component {
  static propTypes = {
    userName: PropTypes.string.isRequired,
  };

  onClick = () => {
    this.userPreferencesModal.openModal();
  };

  render() {
    return (
      <span>
        <Button onClick={this.onClick} bsStyle="success">User preferences</Button>
        <UserPreferencesModal ref={(userPreferencesModal) => { this.userPreferencesModal = userPreferencesModal; }} userName={this.props.userName} />
      </span>
    );
  }
}

export default UserPreferencesButton;
