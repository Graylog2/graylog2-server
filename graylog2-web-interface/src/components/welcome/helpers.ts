export const typeLinkMap = {
  dashboard: { link: 'DASHBOARDS_VIEWID' },
  search: { link: 'SEARCH_VIEWID' },
};

export const DEFAULT_PAGINATION = {
  per_page: 5,
  page: 1,
  count: 0,
  total: 0,
};
