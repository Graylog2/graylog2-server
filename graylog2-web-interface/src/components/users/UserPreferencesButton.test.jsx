import React from 'react';
import { render, fireEvent } from 'wrappedTestingLibrary';

import { StoreMock as MockStore } from 'helpers/mocking';
import UserPreferencesButton from 'components/users/UserPreferencesButton';
import PreferencesStore from 'stores/users/PreferencesStore';

jest.mock('stores/users/PreferencesStore', () => MockStore('get', 'listen', 'loadUserPreferences'));

describe('UserPreferencesButton', () => {
  it('should load user data when user clicks edit button', () => {
    const instance = render(<UserPreferencesButton userName="Full" />);
    const button = instance.getByTestId('user-preferences-button');

    expect(instance.container).toMatchSnapshot();
    expect(button).toBeDefined();

    fireEvent.click(button);

    expect(PreferencesStore.loadUserPreferences).toBeCalled();
  });
});
