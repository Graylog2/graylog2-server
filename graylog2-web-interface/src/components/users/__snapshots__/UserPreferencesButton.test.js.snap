// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferencesButton should load user data when user clicks edit button 1`] = `
<UserPreferencesButton
  userName="Full"
>
  <span>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="success"
      disabled={false}
      onClick={[Function]}
    >
      <button
        className="btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        User preferences
      </button>
    </Button>
    <UserPreferencesModal
      userName="Full"
    >
      <BootstrapModalForm
        cancelButtonText="Cancel"
        formProps={Object {}}
        onSubmitForm={[Function]}
        submitButtonDisabled={false}
        submitButtonText="Save"
        title="Preferences for user Full"
      >
        <BootstrapModalWrapper
          onHide={[Function]}
        >
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <Modal
              autoFocus={true}
              backdrop={true}
              backdropClassName="modal-backdrop"
              backdropTransitionTimeout={150}
              containerClassName="modal-open"
              dialogTransitionTimeout={300}
              enforceFocus={true}
              keyboard={true}
              manager={
                ModalManager {
                  "containers": Array [],
                  "data": Array [],
                  "handleContainerOverflow": true,
                  "hideSiblingNodes": true,
                  "modals": Array [],
                }
              }
              onEntering={[Function]}
              onExited={[Function]}
              onHide={[Function]}
              renderBackdrop={[Function]}
              restoreFocus={true}
              show={false}
              transition={[Function]}
            />
          </Modal>
        </BootstrapModalWrapper>
      </BootstrapModalForm>
    </UserPreferencesModal>
  </span>
</UserPreferencesButton>
`;
